src/main/java/duke/main/Parser.java

```diff
@@ -0,0 +1,167 @@
+        } else if (userInput.startsWith("event")) {
+            String[] info = validateEvent(userInput);
+            return new EventCommand(info[0], info[1], info[2]);
+        } else if (userInput.startsWith("delete")) {
+            int taskIndex = validateMarkAndDelete(userInput.substring(7));
+            return new DeleteCommand(taskIndex);
+        } else if (userInput.startsWith("find")) {
+            String keyword = validateFind(userInput);
+            return new FindCommand(keyword);
+        } else {
+            throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    /**
+     * Validates and extracts the task index from the user input for mark and delete commands.
+     *
+     * @param input The user input containing the task index.
+     * @return The validated task index as an integer.
+     * @throws DukeException If the task index is not valid.
+     */
+    private static int validateMarkAndDelete(String input) throws DukeException {
+        if (input.isEmpty() || !isNumeric(input)) {
+            throw new DukeException("☹ OOPS!!! The index cannot be blank or not an integer.");
+        }
+
+        return Integer.parseInt(input);
+    }
+
+    private static String validateFind(String input) throws DukeException {
+        String keyword = input.replace("find", "");
+        if (keyword.isEmpty()) {
+            throw new DukeException("☹ OOPS!!! Please enter a keyword to find.");
+        }
+
+        return keyword;
+    }
+
+    /**
+     * Checks if a given String can be converted to an integer.
+     *
+     * @param str The String to be checked.
+     * @return True if the String can be converted to an integer, false otherwise.
+     */
+    private static boolean isNumeric(String str) {
+        if (str == null || str == "") {
+            return false;
+        }
+        try {
+            Integer.parseInt(str);
+            return true;
+        } catch (NumberFormatException e) {
+            return false;
+        }
+    }
+
+    /**
+     * Validates a toDo description to ensure it is not empty.
+     *
+     * @param todo The toDo description to be validated.
+     * @throws DukeException If the toDo description is empty.
+     */
+    private static void validateToDo(String todo) throws DukeException {
+        if (todo.isEmpty()) {
+            throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+        }
+    }
+
+    /**
+     * Validates and extracts information from user input for creating a deadline.
+     *
+     * @param input The user input for creating a deadline.
+     * @return A String array containing the task description and deadline information.
+     * @throws DukeException If the input format is invalid.
+     */
+    private static String[] validateDeadline(String input) throws DukeException {
+        String[] splitInput = input.replace("deadline", "").split(" /by ");
```

Comment:
Replacing the string might cause unexpected issues, eg. when "deadline" is part of the name of the Deadline Task. This can be seen in the validateEvent as well